src/terminal/terminal.c

```diff
@@ -847,6 +847,45 @@ int guac_terminal_clear_range(guac_terminal* term,
 
 }
 
+/**
+ * Returns whether the given character would be visible relative to the
+ * background of the given terminal.
+ *
+ * @param term
+ *     The guac_terminal to test the character against.
+ *
+ * @param c
+ *     The character being tested.
+ *
+ * @return
+ *     true if the given character is different from the terminal background,
+ *     false otherwise.
+ */
+static bool guac_terminal_is_visible(guac_terminal* term,
+        guac_terminal_char* c) {
+
+    /* Continuation characters are NEVER visible */
+    if (c->value == GUAC_CHAR_CONTINUATION)
+        return false;
+
+    /* Characters with glyphs are ALWAYS visible */
+    if (guac_terminal_has_glyph(c->value))
+        return true;
+
+    int background;
+
+    /* Determine actual background color of character */
+    if (c->attributes.reverse != c->attributes.cursor)
```

Comment:
I'm not sure I understand what this check is about - comparing the reverse attribute to the cursor attribute?